lib/api/manipulation.js

```diff
@@ -410,23 +410,27 @@ exports.wrapAll = function (wrapper) {
       wrapper = wrapper.call(this[0]);
     }
 
-    var wrap = this._make(wrapper);
+    // trim whitespace
+    if (typeof wrapper === 'string') {
+      wrapper = wrapper.trim();
+    }
+
+    var wrap = this._make(wrapper).eq(0).clone();
 
     if (this[0].parent) {
       wrap = wrap.insertBefore(this[0]);
     }
 
-    // if html is given as wrapper, wrap may contain text elements
-    var elInsertLocation = { children: wrap };
+    var elInsertLocation = wrap[0];
     var j = 0;
 
     // Find the deepest child. Only consider the first tag child of each node
     // (ignore text); stop if no children are found.
-    while (
-      elInsertLocation &&
-      elInsertLocation.children &&
-      j >= elInsertLocation.children.length
-    ) {
+    while (elInsertLocation && elInsertLocation.children) {
```

Comment:
Whee, forgot to flip the order of the `if` when including it in the loop condition. Could you add a test case that breaks without this change? Also include it in the `while` condition, as it makes the code more readable?

Reply:
All code here is actually letter by letter from `_wrap` function.

Reply:
I saw that — I've included this change in #1596, as tests broke when I migrated to Jest. Looks like tests weren't run properly on Mocha.

Reply:
Actually tests did fail. In tests I used nested elements and check are elements attached in correct position.